logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/ProteusClientProvider.kt

```diff
@@ -58,7 +60,8 @@ class ProteusClientProviderImpl(
     private val userId: UserId,
     private val passphraseStorage: PassphraseStorage,
     private val kaliumConfigs: KaliumConfigs,
-    private val dispatcher: KaliumDispatcher = KaliumDispatcherImpl
+    private val dispatcher: KaliumDispatcher = KaliumDispatcherImpl,
+    private val proteusMigrationRecoveryHandler: ProteusMigrationRecoveryHandler
```

Comment:
Is it possible to write some tests for this class?

Reply:
We didn't have before and I added it on this PR, It was just bad naming from my side (kept the `Impl`) suffix, fixed.